lms/djangoapps/django_comment_client/utils.py

```diff
@@ -151,6 +123,20 @@ def get_discussion_id_map_entry(xblock):
     )
 
 
+@request_cached
+def get_accessible_discussion_xblocks_by_course_id(course_id, user=None, include_all=False):
+    """
+    Return a list of all valid discussion xblocks in this course.
+    Checks for the given user's access if include_all is False.
+    """
+    all_blocks = get_discussion_xblocks_by_course_id(course_id)
```

Comment:
Splitting this method into two parts (access checks here in the LMS process, `get_discussion_xblocks_by_course_id` in common) is one of the few actual changes to this code.